### DISCUSSION TITLE
When testing a reflex web app, is there a way to select a file if the rx.upload() component is used to select files to upload?

### DISCUSSION THREAD

**Discussion by aden-collector** [reflex-dev/reflex#3323] [General]
I've been able to write pytest python code to test a reflex web app.  I use the AppHarness module and am able to locate elements on a reflex web page and manipulate them (click object, etc.).  I am trying to test uploading of files on the app.  The app uses the rx.upload() component to select the files to upload.  I am unable to use a web component identifier (xpath, etc.) to select a file.  Are there any workarounds to select a file for testing like triggering an event or running a javas script function?

**Comment by masenf** [comment]
Check out how we do it in `integration/test_upload.py`

https://github.com/reflex-dev/reflex/blob/99d59104ad71b6dd28426b14ff5721c685a5bf5d/integration/test_upload.py#L180-L182

https://github.com/reflex-dev/reflex/blob/99d59104ad71b6dd28426b14ff5721c685a5bf5d/integration/test_upload.py#L192
